DBTools/python/getSiblings.py

```diff
@@ -113,13 +131,193 @@ def setAPIs(self):
         elif args.prod == 'allUser':
             self.dbsapi_in = DbsApi (url = 'https://cmsweb.cern.ch/dbs/prod/phys03/DBSReader')
             self.dbsapi_out = DbsApi (url = 'https://cmsweb.cern.ch/dbs/prod/phys03/DBSReader')
+    '''
 
     #main function for getting siblings using either lumi matching or DBS
     def getSiblings(self):
         if args.prod == 'local' or args.prod == 'allLocal' or args.lumiMatching:
-            self.getLumiMatching(self.dictName, args.prod, args.inputJSON, args.siblingJSON)
+            #self.getLumiMatching(self.dictName, args.prod, args.inputJSON, args.siblingJSON)
+            print("Running local option...")
+            self.local = True
+            self.getFilesFromList(args.jobNumber, args.totalJobs)
+            self.findMatches()
         else:
-            self.getDBSSiblings(self.dictName)
+            self.getFilesFromList(args.jobNumber, args.totalJobs)
+            self.findMatches()
+
+    @staticmethod
+    def getDASInfo(dataset, jsonName=None):
+        cmd = 'dasgoclient -query="file,run,lumi,events dataset={0}" -json'.format(dataset)
+        process = subprocess.Popen(cmd, stdout=subprocess.PIPE, shell=True)
+        output, error = process.communicate()
+
+        miniaod = output.decode()
+
+        test = ast.literal_eval(miniaod)
+        files = [x['file'][0]['name'] for x in test]
+        lumis = [[x['lumi'][0]['number']] if isinstance(x['lumi'][0]['number'], int) else x['lumi'][0]['number'] for x in test]
+        runs = [[x['run'][0]['run_number']] if isinstance(x['run'][0]['run_number'], int) else x['run'][0]['run_number'] for x in test]
+
+        miniDict = {}
+
+        for f, l, r in zip(files, lumis, runs):
+            miniDict[f] = {'lumis': l, 'runs': r}
+
+        if jsonName:
+            with open(jsonName, 'w') as fout:
+                json.dump(miniDict, fout)
+
+        return miniDict
+
+    @staticmethod
+    def getLocalInfo(dataset, jsonName=None):
+
+        miniDict = {}
+
+        for filename in os.listdir(dataset):
+            if not filename.endswith('.root'): continue
+            print("Working on file", dataset+filename)
+            lumi_list = []
+            run_list = []
+
+            fin = r.TFile.Open(dataset + filename, 'read')
```

Comment:
Given the issue we were seeing with the local files, and since this is a "new" way of accessing it, this might be the source of illegal instructions. Not sure why or how, but it is the only distinction from the previous getSiblings. I would suggest to use the `r.getEventsInFile()` here to get the lumis (also to avoid the memory issues we saw a while ago, which are not very relevant for the local case, but still)